KomaMRIBase/src/datatypes/Phantom.jl

```diff
@@ -400,36 +454,59 @@ end
     obj = pelvis_phantom2D(; ss=4, us=1)
 
 Creates a two-dimensional pelvis Phantom struct.
+Default ss=4 sample spacing is 2 mm. Original file (ss=1) sample spacing is .5 mm.
 
 # Keywords
-- `ss`: (`::Integer`, `=4`) subsampling parameter
-- `us`: (`::Integer`, `=1`) upsampling parameter for all axes, if used ss is set to ss=1
+- `ss`: (`::Integer or ::Vector{Integer}`, `=4`) subsampling parameter for all axes if scaler, per axis if 2 element vector [ssx, ssy]
+- `us`: (`::Integer or ::Vector{Integer}`, `=1`)  upsampling parameter for all axes if scaler, per axis if 2 element vector [usx, usy]
 
 # Returns
 - `obj`: (`::Phantom`) Phantom struct
 
 # Examples
 ```julia-repl
-julia> obj = pelvis_phantom2D(; ss=1)
+julia> obj = pelvis_phantom2D(; ss=2])
+
+julia> obj = pelvis_phantom2D(; us=[1, 2])
 
 julia> pelvis_phantom2D(obj, :ρ)
 ```
 """
 function pelvis_phantom2D(; ss=4, us=1)
 
+    # check for valid input    
+    @assert length( ss) <= 2 "ss=$(ss) invalid, ss can have up to two components for a 2D phantom"
+    @assert length( us) <= 2 "us=$(us) invalid, us can have up to two components for a 2D phantom"
+    if length( us) > 1 || prod( us) > 1
+        @info "setting ss=1 since us=$(us) defined"
+        ss = 1
+    end
+    if length( us) == 1
+        usx = us[1]; usy = us[1]
+    else
+        usx = us[1]; usy = us[2]
+    end
+    if length( ss) == 1
+        ssx = ss[1]; ssy = ss[1]
+    else
+        ssx = ss[1]; ssy = ss[2]
+    end
```

Comment:
As we are using this code multiple times, maybe we can put it in a function `check_phantom_sampling!(ss, us)` that modifies `us` and `ss` directly, so we can always use `us[1]`...`us[3]` and  `ss[1]`...`ss[3]`.

Reply:
That seems reasonable, especially if more phantoms are defined going forward.

Possibly some more of the code could go in supporting functions or macros going forward as well. A possible variation would be a an abstract phantom parent object inheriting the dimensionality and the specific input dataset along with supporting methods?

At some point this would be interesting to consider along with incorporating "big brain"?

Reply:
> Possibly some more of the code could go in supporting functions or macros going forward as well. A possible variation would be a an abstract phantom parent object inheriting the dimensionality and the specific input dataset along with supporting methods?

I am not sure if I understood correctly, but right now the maps (`obj.T1`, `obj.T2`, etc..) are converted to `Vector{T}`'s before being stored in `Phantom{T}`, so we lose some information, your suggestion would be to enable `Array{T,N}`s? If that is the case, that sounds good to me. 

It was like that at some point actually. I simplified it for simulation reasons, so the first dimension is the "spin" dimension, and the second dimension is time. Nevertheless, there are no good reasons (besides simulating fewer points ` ρ.==0`) to pass this burden to the user, and in the cases in which we have `Phantom{Array{T,N}}`, that has an implicit structure, the spins' positions could be dropped.

Reply:
Saw the #347. I can contribute to dicussion and coding (once the ideas are layed out) as needed. Very interested in motion as well.